### PR TITLE
Addressable but up to date

### PR DESCRIPTION
**Pull request by Bod9001**
just need to make the asset picker support multiple projects 

I'll do a proper write-up tomorrow

ok, in the editor
in the editor dynamically looks for the projects with the built bundles in 
unitystation\UnityProject\AddressablePackingProjects
and looks for folder called ServerData that it expects to find the catalogued in, noting that the paths inside the catalogue are set up for the editor, 

When you do a dev build it changes the paths and moves unitystation\UnityProject\AddressablePackingProjects into streaming assets and changes the paths in the catalogue with new positions,,

 for builds it uses the config to Load them from the Internet note need a process for changing the addresses to web addresses for the bundles in the catalogue

still need to do a non-play mode addressable for sprites and stuff, I was getting an error when I tried it

note
sounds need to be done, the only sound that works currently is the UI action one

### CODE REVIEW

**Comment by Bod9001** [comment]
dam github desktop rebase is so messed up, It completely broke my branch xD

**Comment by Bod9001** [comment]
there we go fixed

File: UnityProject/Assets/Resources/Prefabs/SceneConstruction/NestedManagers/ScriptableObjectSingletons.prefab [hunk 1]
```diff
@@ -57,7 +57,6 @@ MonoBehaviour:
     type: 2}
   CommonPrefabs: {fileID: 11400000, guid: bf1ea2a89b19143c996995d650266ff2, type: 2}
   CommonCooldowns: {fileID: 11400000, guid: 5b45f6c0f9827844aa17cd49d2e888f1, type: 2}
-  AmmoPrefabs: {fileID: 11400000, guid: 80457940d6023324eb9a2f1fc72d6a49, type: 2}
```
Comment (NoooneyDude) [inline UnityProject/Assets/Resources/Prefabs/SceneConstruction/NestedManagers/ScriptableObjectSingletons.prefab:60]: `AmmoPrefabs` was removed. Is that intended?

File: UnityProject/Assets/Scripts/Core/Addressables/CatalogLoader.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        Debug.Log("locationsLoaded ==> " + obj.Status);
+        if (obj.Status == AsyncOperationStatus.Succeeded)
+        {
+            locations = new List<IResourceLocation>(obj.Result);
+            loadDependency();
+        }
+        else
+        {
+            Debug.LogError("locationsLoaded is failed");
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Addressables/CatalogLoader.cs]: There's a bunch of `Debug.Log` calls in this file. Perhaps use `Logger.Log`, `Category.Addressables`?

File: UnityProject/Assets/Scripts/Core/Addressables/TestAddressableScript.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.AddressableAssets;
+
+public class TestAddressableScript : MonoBehaviour
+{
+
+	//[NaughtyAttributes.Button]
+	//[MenuItem("Tools/LoadCatalog")]
+	//public static void LoadCatalog()
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Addressables/TestAddressableScript.cs]: Everything potentially useful in this file is commented out. Could this file be deleted?

File: UnityProject/Assets/Scripts/Core/Addressables/Types/TestingQuickly.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    void Update()
+    {
+
+    }
+
+	[RightClickMethod]
+    [NaughtyAttributes.Button("NotValidKey?")]
+    public void bob()
+    {
+	    //Reference.NotValidKey();
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Addressables/Types/TestingQuickly.cs]: Same thing here, and there are empty `MonoBehaviour` methods. Could this be removed, too?

File: UnityProject/Assets/Scripts/Core/Editor/Addressable editor/AddressablesInEditor.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+using UnityEngine.ResourceManagement.AsyncOperations;
+using UnityEngine.ResourceManagement.ResourceLocations;
+
+
+public class AddressablesInEditor : EditorWindow
+{
+	public static List<IResourceLocation> locations = new List<IResourceLocation>();
+
+	// [MenuItem("Tools/LoadCatalog")]
+	// public static void LoadCatalog()
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Editor/Addressable editor/AddressablesInEditor.cs]: Looks like this main menu method is commented out, is this another test file that could be deleted?

**Comment by NoooneyDude** [comment]
Some other things I noticed:
- Includes changes from PRs #5670 and #5674, which are already merged - so not sure why GitHub is showing duplicate changes?
- `AddressablePackingProjects/SoundAndMusic/Assets/empty.unity`. Is this needed? Looks like a default scene or something.
- A few //JESTER SoundManager references are still commented out:
    - `WindowInteract.cs` - 33
    - `AdminCommandsManager.cs` - 343
    - `GameManager.cs` - 458
    - `PlayerSync.Server.cs` - 593
- A bunch of new sound files are removed. These seem to be new files added since the start of sound addressables. See the files changed tab, very close to the bottom for the list of sounds.

File: UnityProject/Assets/Resources/Prefabs/SceneConstruction/NestedManagers/ScriptableObjectSingletons.prefab (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Resources/Prefabs/SceneConstruction/NestedManagers/ScriptableObjectSingletons.prefab:60]: yeah, accidental

File: UnityProject/Assets/Scripts/Core/Addressables/CatalogLoader.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Addressables/CatalogLoader.cs]: yeah, that's from Danzer

File: UnityProject/Assets/Scripts/Core/Addressables/TestAddressableScript.cs (hunk 3 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Addressables/TestAddressableScript.cs]: will clean

File: UnityProject/Assets/Scripts/Core/Addressables/Types/TestingQuickly.cs (hunk 4 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Addressables/Types/TestingQuickly.cs]: will clean

File: UnityProject/Assets/Scripts/Core/Editor/Addressable editor/AddressablesInEditor.cs (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Editor/Addressable editor/AddressablesInEditor.cs]: yeah

File: UnityProject/Assets/Resources/Prefabs/SceneConstruction/NestedManagers/ScriptableObjectSingletons.prefab (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Resources/Prefabs/SceneConstruction/NestedManagers/ScriptableObjectSingletons.prefab:60]: looks like it was removed in a previous PR so the prefab wasn't updated

**Comment by Bod9001** [comment]
AddressablePackingProjects/SoundAndMusic/Assets/empty.unity
is so when you hit control S it has something to save otherwise a try saving the default scene, and it will make a new scene for it

**Comment by Bod9001** [comment]
"A few //JESTER SoundManager references are still commented out: "
those stuff are more nuanced and require some Extra Scripts, to get working and I had to pull out the relevant files and bring them up to an update branch to get it updated so a lot of stuff got lost in the way

**Comment by Bod9001** [comment]
let's see if this works

**Pull request merged**